ntrans_gui.py

```diff
@@ -262,21 +305,30 @@ def update_estimated_time_label(self, *junk: object) -> None:
         total_time_in_seconds = int(
             self.data_size_var.get()
             * len([n for n, var in self.checkbox_vars.items() if var.get()])
-            * 1.2  # Average time, sec per string
-        )
-        if total_time_in_seconds >= 60:
+            * 1.2
+        )  # Average time in seconds per N-gram
+        if total_time_in_seconds >= 60 * 60:
+            self.estimated_time_label[
+                "text"
+            ] = f"Estimated run time: {int(total_time_in_seconds / 60 / 60)} h & {int(total_time_in_seconds / 60 % 60)} min"
+        elif total_time_in_seconds >= 60:
             if total_time_in_seconds % 60 == 0:
                 self.estimated_time_label[
                     "text"
-                ] = f"Estimated time: {int(total_time_in_seconds / 60)} min"
+                ] = f"Estimated run time: {int(total_time_in_seconds / 60)} min"
             else:
                 self.estimated_time_label[
                     "text"
-                ] = f"Estimated time: {int(total_time_in_seconds / 60)} min & {total_time_in_seconds % 60} sec"
+                ] = f"Estimated run time: {int(total_time_in_seconds / 60)} min & {total_time_in_seconds % 60} sec"
         else:
             self.estimated_time_label[
                 "text"
-            ] = f"Estimated time: {total_time_in_seconds} sec"
+            ] = f"Estimated run time: {total_time_in_seconds} sec"
+
+    def cancel_generation(self) -> None:
+        self.cancel_thread_event.set()
+        self.thread = None
+        self.cancel_thread_event.clear()
```

Comment:
This doesn't seem right. The event is set for a very short time until it's cleared again. Are you sure this works?

Reply:
Maybe `self.cancel_thread_event.clear()` should be just before making a new thread?